pkg/kubectl/cmd/delete.go

```diff
@@ -116,6 +116,7 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		SelectAllParam(deleteAll).
 		ResourceTypeOrNameArgs(false, args...).RequireObject(false).
 		Flatten().
+		Latest().
```

Comment:
Yes, there is a flag, but I'm trying to solve a completely different problem: nodecontroller and apiserver may enter into a race to delete a pod when `kubectl delete pod/name --grace-period=0` is executed. The end result is kubectl may return a NotFound error while it actually is the kubectl deletes the pod. By forcing `kubectl delete` to always do a "GET" and then a "DELETE", kubectl can safely ignore the NotFound error for the "DELETE" request and thus solve the bug.